### ISSUE TITLE
GenericResponseError 9020 From EWS

### ISSUE THREAD

**Issue by tgelphman** [microsoft/MHA#80]
Can you please assist me in uninstalling MHA from outlook 2016. I am not connected to an office 365 account. It is a standalone client connected to my corp on-prem exchange server. I see no way to uninstall MHA. As you can see, it is eroring out when i try to use it.
![mha error](https://user-images.githubusercontent.com/38364672/38759209-920e0e4a-3f29-11e8-8050-60fc6f2bd2f5.png)

**Comment by stephenegriffin** [comment]
I’m guessing your administrator installed it and enabled it for you. You might be able to disable it through the store icon, and if not, your internal help desk might be able to assist. That error you’re getting is one I’m very interested in tracking down though. Is there any chance you’d be willing/able to gather some more data to help me understand why it’s happening?

**Comment by tgelphman** [comment]
Hi and thank you for answering. As for who installed it, that would be me. I’m the IT Manager and do all things, including installing software on my own machine. It has been a while since I installed it so I don’t even remember how I did .. whether as an add-on or standalone installation package. I’m guessing you’d know how you distributed it. So, the long answer is I’m the guy who needs to figure out how to uninstall it ☺.

And, yes, I’d be more than happy to assist you in troubleshooting. As a software developer myself, I am always glad to help others. Tell me what you would like me to do and what data to get you. Naturally I’ll make sure none of it is data I’d consider internal only.

T
---
Tony Gelphman
Manager of Information Technology
City of Redwood City<http://www.redwoodcity.org/finance/it/index.html>
(650) 780-7090
<removed message being replied to>

**Comment by tgelphman** [comment]
I meant to add, I’d actually prefer to get it working rather than uninstall it. It was quite valuable when working in diagnosing spam and other illegitimate email. My reason for uninstalling was thinking I’d uninstall and then reinstall. However, not only was I unable to uninstall, I was unable to find a version to reinstall as the links on your site kept trying to add it to my OWA vs my outlook client connecting to our internal exchange server.

T
<removed message being replied to>

**Comment by stephenegriffin** [comment]
It's an Office Store app, so you manage it through the Store icon. What I'd like to see is a Fiddler trace of the app running. DM me over on [twitter](https://twitter.com/sgriffin) and we can work out the details.

**Comment by stephenegriffin** [comment]
Documentation:
https://dev.office.com/reference/add-ins/outlook/preview/Office.context.mailbox.item?product=outlook
https://docs.microsoft.com/en-us/outlook/actionable-messages/invoke-add-in-from-actionable-message

Related issues:
#78
https://github.com/OfficeDev/Outlook-Add-in-JavaScript-MakeEWSRequest/issues/1
https://social.msdn.microsoft.com/Forums/security/en-US/9e35d385-1afa-41a1-9ffc-0d755b190ee3/makeewsrequestasync-function-in-outook-2016-windows-desktop-client-doesnt-work?forum=appsforoffice
https://stackoverflow.com/questions/44566134/office-addin-load-custom-properties-on-saved-events

**Comment by stephenegriffin** [comment]
This is the same issue as #78

**Comment by stephenegriffin** [comment]
Aha - 9020 is from outlook-*.js - the call is coming from inside the house:
$h.OutlookErrorManager.OutlookErrorCodes.genericResponseError=9020;
$h.OutlookErrorManager._addErrorMessage$p(9020,"GenericResponseError","");

Places which log 9020:
1. _loadDictionaryRequest
2. SaveDictionaryRequest - used by saveAsync
I use both of the above, but we're getting this error from an EWS request, which is the third place this is logged:
3. handleResponse - when wasProxySuccessful is false.

Which leads to these possibly related reports:
https://www.veritas.com/support/en_US/article.100031398
https://www.veritas.com/support/en_US/article.100033597
https://vox.veritas.com/t5/Enterprise-Vault/Failed-to-send-a-request-to-the-Exchange-Server-OWA-2013/td-p/670279

And this tantalizing blog post which describes the proxy in more detail:
https://blogs.msdn.microsoft.com/emeamsgdev/2014/09/19/troubleshooting-exchange-2013-mail-apps-executeewsproxy-calls/

Possible conclusion: This error is due to misconfig on the Exchange server.
Possible fix: Put up a page describing how to troubleshoot and fix misconfig, then redirect users getting 9020 to this page.

**Comment by stephenegriffin** [comment]
@tgelphman - per my previous post - could you check your EWL url config (and possibly walk through troubleshooting steps given in [Andrei's blog](https://blogs.msdn.microsoft.com/emeamsgdev/2014/09/19/troubleshooting-exchange-2013-mail-apps-executeewsproxy-calls/) and see if you can:

1. Confirm you're getting wasProxySuccessful == false
2. Address the issue from your side through config.

**Comment by tgelphman** [comment]
Hi Stephen, I see a few emails in which you point me stating they are the same issue; however, in none of them do I see the fix. I would first like to just do something simple, uninstall and reinstall … experience proves that fixes 90% of issues … before digging into the logs. Can you please explain to me how to uninstall the add-in? I do not see it listed in add-ins or in control panel/programs. I know I didn’t install it doing anything on our exchange server so it is all local. Also, once uninstalled, do you have a link directly to where I can get the code to reinstall? When I tried to find that it took me to where I could install for an office 365 implementation, but not a client/server based install with on-prem Exchange and client office 2016.


T
---
Tony Gelphman
Manager of Information Technology
(650) 780-7090 Office

Computers are incredibly fast, accurate and stupid. Human beings are incredibly slow, inaccurate and brilliant. Together they are powerful beyond imagination.  Einstein.
[removed previous replies]

**Comment by stephenegriffin** [comment]
>It's an Office Store app, so you manage it through the Store icon.

This red shopping bag icon with Store and Add-ins written underneath it is where you manage this kind of add-in.
![image](https://user-images.githubusercontent.com/6773183/38887646-69e3310e-4247-11e8-9dba-793720b3e85e.png)
The whole experience is managed through the Store and OWA/Exchange admin portal  and I have no control over it. If you do manage to remove the add-in, you should be able to locate it again in the store, or else here's the [direct link](https://appsource.microsoft.com/en-us/product/office/WA104005406?src=office&corrid=eeaf65c8-bf39-4ab2-9096-33bbe1672384&omexanonuid=f7cbeeba-7e18-4118-b5cc-e6ca4d266254).

I can tell you that this is NOT an installation problem and it will NOT be resolved by uninstalling and reinstalling, because there's nothing actually "installed" here in the first place. All you're doing in the store is opting in to a manifest which tells Outlook what URL to go to to load the web page that is MHA. Every time you open the add-in a new copy of the code is fetched from my server.

>in none of them do I see the fix

I don't think I suggested any of them had "the fix" in them. Just that they were other people reporting the same issue in their code, which is useful information in trying to work out the cause of a problem. I strongly suspect the problem here is config on your end, which is why I wanted you to go through [Andrei's blog](https://blogs.msdn.microsoft.com/emeamsgdev/2014/09/19/troubleshooting-exchange-2013-mail-apps-executeewsproxy-calls/) to see if you could narrow down where the problem is.

**Comment by tgelphman** [comment]
Well as much as I liked the addin, being unable to uninstall it has become a pain, forcing me basically to uninstall and reinstall office itself. When I try to do access it from the Store, it asks for my Office account, then tries to find an associated O365 account, which it never will since I do not have one. I am not sure if this is a Microsoft thing or not; however, from my perspective it is just a problem that I need to remove any way I can. Thank you though for your attempt to help me. Bummer though it didn’t work as I was considering having it installed on all 500 PCs under my purview as a tool to detect and fight spam.


T
---
Tony Gelphman
Manager of Information Technology
(650) 780-7090 Office

Computers are incredibly fast, accurate and stupid. Human beings are incredibly slow, inaccurate and brilliant. Together they are powerful beyond imagination.  Einstein.

[removed previous replies]

**Comment by stephenegriffin** [comment]
The add-in is installed to either your Exchange server or your mailbox. It's not installed to Outlook. So re-installing Outlook wouldn't do anything. Maybe you installed MHA via your [Exchange Admin Center](https://technet.microsoft.com/en-us/library/jj150562(v=exchg.160).aspx), perhaps even using a manifest file. 

Here are the steps from the front page of this GitHub on managing an add-in via the EAC. You should be able to reverse them:
- Go to the Exchange Admin Center.
- Click on the Organization tab
- From there, select the add-ins tab
- Click the Plus icon/Add from the Office Store
- A new page will load for the store
- Search for "Message Header Analyzer"
- Choose MHA in the results
- Click Add
- Confirm by clicking Yes
- Back in the Exchange Admin Center, refresh the list of add-ins
- You can now edit who the add-in is available for

>Thank you though for your attempt to help me.

I do still want to figure this out, but I need data to help you. If you ever get a fiddler trace of the error, I would love to look at it.

**Comment by tgelphman** [comment]
Stephen, I respectfully disagree in that I did install it locally only, never touching my exchange server. This was verified over the last hour when I uninstalled Office and then reinstalled. Once done, MHA was gone. I did follow your instructions below just to further verify if there was a footprint of any kind in Exchange and it was nowhere to be found. I wish I could tell you how I installed it locally only but it’s been a while and I just can’t remember.

In any event, I appreciate your support and response. Too often I work with vendors/software developers who just do not understand support is part of the equation when developing software … it doesn’t just end with a release.

Have a great day,

T
---
Tony Gelphman
Manager of Information Technology
(650) 780-7090 Office

Computers are incredibly fast, accurate and stupid. Human beings are incredibly slow, inaccurate and brilliant. Together they are powerful beyond imagination.  Einstein.

[previous replies removed]
